catalog/models.py

```diff
@@ -38,6 +41,13 @@ class BookInstance(models.Model):
     book = models.ForeignKey('Book', on_delete=models.RESTRICT, null=True)
     imprint = models.CharField(max_length=200)
     due_back = models.DateField(null=True, blank=True)
+    borrower = models.ForeignKey(User, on_delete=models.SET_NULL, null=True, blank=True)
```

Comment:
option `on_delete` có những giá trị nào? ý nghĩa mỗi giá trị đó

Reply:
Dạ option on_delete có 6 giá trị:
CASCADE: Khi đối tượng được tham chiếu bị xóa thi sẽ xóa luôn các đối tượng có tham chiếu đến nó.
PROTECT: Cấm xóa đối tượng được tham chiếu. Để xóa nó, sẽ phải xóa tất cả các đối tượng tham chiếu thủ công. 
SET_NULL: Đặt tham chiếu thành NULL (yêu cầu trường là nullable).
SET_DEFAULT: Đặt giá trị mặc định. 
SET(...): Đặt một giá trị nhất định. 
DO_NOTHING: Tạo ra các vấn đề toàn vẹn trong cơ sở dữ liệu của bạn (tham chiếu một đối tượng thực sự không tồn tại). Tương đương SQL : NO ACTION.

Reply:
So sánh `RESTRICT` với `PROTECT`